### DISCUSSION TITLE
Configure remote BitbucketServer on a sub-path?

### DISCUSSION THREAD

**Discussion by GriceTurrble** [gitkraken/vscode-gitlens#1382] [Q&A]
I'm trying to configure GitLens' `remotes` setting to connect to our company BitbucketServer instance. It seems to work fine if I just give it `"domain": "company-domain.com"` and `"type": "BitbucketServer"`; but in our case we have multiple services on sub-paths of that domain. For instance:

- Jira: `company-domain.com/jira`
- BitbucketServer: `company-domain.com/bitbucket`
- Confluence: `company-domain.com/confluence`

There are other tools we have that are all packaged as a single entity, like Jenkins at `company-domain.com/jenkins`, and IT has done quite a bit of work getting them all talking to each other; but I digress.

So, if I configure GitLens with `"domain": "company-domain.com/bitbucket"`, it doesn't even recognize that that config exists (probably fails somewhere internally, since that's not strictly a domain). Instead, it appears I need to configure a custom remote and specify all the URL patterns manually.

Is there an option I'm missing that can set the path to the root of our BitbucketServer instance? Or am I stuck writing it as a custom config?

Thank you.

**Comment by eamodio** [comment]
You should be able to use the new `regex` property to configure the remote better (make sure to capture the "domain" part and the "path" part, e.g.
```
 { "type": "BitbucketServer", "regex": "^(.*/bitbucket)/scm/(.*)$" }
```

Hope that helps.

**Reply by GriceTurrble** [reply]
Had to play around with the regex option a bit, but got it to at least recognize the remote. Now I've got a different problem.

I have my remotes set up via SSH, and that path is slightly different. The web portal is something like `https://server.name/bitbucket/...`, while the SSH remote is in the form `ssh://git@server.name/...` (no extra path there). So I don't think it's possible to both capture the SSH path and rework it to the correct HTTPS path within just the regex. 

It might be helpful here if we could use the regex to identify the remote, which I can craft to use both HTTPS and SSH styles as needed on my end; then provide a `baseUrl` option that just has `https://server.name/bitbucket`. GitLens can use that as an override, ignoring both regex and domain; and appending the rest of the paths (that it already knows how to format) on top of that.

Heck, in that regard, just saying `domain` and `baseUrl` would be enough, I think: match the remote by domain, then tell it which baseUrl to add to formatted remote paths.

**Reply by eamodio** [reply]
Yeah, there is an open feature request here for that: https://github.com/eamodio/vscode-gitlens/issues/1364.
